regression in build on macos

Since https://github.com/ros2/ros2_tracing/pull/31 (I think it's this one) this no longer builds on macos:

```
% colcon build --symlink-install --merge-install --cmake-args -DPython3_EXECUTABLE=`which python3` -DPYTHON_EXECUTABLE=`which python3` --mixin compile-commands debug --event-handlers console_direct+ console_stderr- --packages-select tracetools
Starting >>> tracetools
Not searching for unused variables given on the command line.
-- Found ament_cmake_ros: 0.11.2 (/Users/william/2011_ws/install/share/ament_cmake_ros/cmake)
-- Override CMake install command with custom implementation using symlinks instead of copying resources
-- Checking for module 'lttng-ust'
--   No package 'lttng-ust' found
CMake Error at /opt/homebrew/Cellar/cmake/3.23.2/share/cmake/Modules/FindPkgConfig.cmake:605 (message):
  A required package was not found
Call Stack (most recent call first):
  /opt/homebrew/Cellar/cmake/3.23.2/share/cmake/Modules/FindPkgConfig.cmake:827 (_pkg_check_modules_internal)
  CMakeLists.txt:36 (pkg_check_modules)


-- Configuring incomplete, errors occurred!
See also "/Users/william/2011_ws/build/tracetools/CMakeFiles/CMakeOutput.log".
See also "/Users/william/2011_ws/build/tracetools/CMakeFiles/CMakeError.log".
Failed   <<< tracetools [0.45s, exited with code 1]

Summary: 0 packages finished [1.09s]
  1 package failed: tracetools
  1 package had stderr output: tracetools
```

Reverting to e664e5b let me build cleanly (with some compiler warnings):

```
% colcon build --symlink-install --merge-install --cmake-args -DPython3_EXECUTABLE=`which python3` -DPYTHON_EXECUTABLE=`which python3` --mixin compile-commands debug --event-handlers console_direct+ console_stderr- --packages-select tracetools
Starting >>> tracetools
Not searching for unused variables given on the command line.
-- Found ament_cmake_ros: 0.11.2 (/Users/william/2011_ws/install/share/ament_cmake_ros/cmake)
-- Override CMake install command with custom implementation using symlinks instead of copying resources
-- Checking for module 'lttng-ust'
--   No package 'lttng-ust' found
-- Found ament_lint_auto: 0.14.0 (/Users/william/2011_ws/install/share/ament_lint_auto/cmake)
-- Found gtest sources under '/Users/william/2011_ws/install/src/gtest_vendor': C++ tests using 'Google Test' will be built
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Added test 'copyright' to check source files copyright and LICENSE
-- Added test 'cppcheck' to perform static code analysis on C / C++ code
-- Configured cppcheck include dirs:
-- Configured cppcheck exclude dirs and/or files:
-- Added test 'cpplint' to check C / C++ code against the Google style
-- Configured cpplint exclude dirs and/or files:
-- Added test 'lint_cmake' to check CMake code style
-- Added test 'uncrustify' to check C / C++ code style
-- Configured uncrustify additional arguments:
-- Added test 'xmllint' to check XML markup files
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/william/2011_ws/build/tracetools
[ 23%] Building CXX object gtest/CMakeFiles/gtest.dir/src/gtest-all.cc.o
[ 23%] Building C object CMakeFiles/tracetools.dir/src/tracetools.c.o
[ 23%] Building CXX object CMakeFiles/tracetools.dir/src/utils.cpp.o
...
```

I know macos isn't a tiered platform anymore, but it would be nice to have it compile still. I briefly glance at the pr, but I wasn't able to see what should change quickly, so I decided to just open an issue for now.

--- a/test_tracetools/CMakeLists.txt
+++ b/test_tracetools/CMakeLists.txt
@@ -13,7 +13,7 @@ endif()
 
 find_package(ament_cmake REQUIRED)
 
-if(WIN32)
+if(WIN32 OR APPLE)
   set(DISABLED_DEFAULT ON)
 else()
   set(DISABLED_DEFAULT OFF)

--- a/tracetools/CMakeLists.txt
+++ b/tracetools/CMakeLists.txt
@@ -15,7 +15,7 @@ endif()
 
 find_package(ament_cmake_ros REQUIRED)
 
-if(WIN32)
+if(WIN32 OR APPLE)
   set(DISABLED_DEFAULT ON)
   set(STATUS_CHECKING_TOOL_DEFAULT OFF)
 else()

